### PR TITLE
Fixup/sqlbugs

### PR DESCRIPTION
**Pull request by rabble**
This makes SQL migrations more robust which catches some errors users are seeing.

### CODE REVIEW

**Comment by martindsq** [comment]
@rabble What about this PR, should we merge it?

File: Source/GoBot/ViewDatabase.swift [hunk 1]
```diff
@@ -232,33 +232,33 @@ class ViewDatabase {
...
+                CREATE INDEX IF NOT EXISTS messagekeys_id ON messagekeys(id);
+                CREATE INDEX IF NOT EXISTS posts_msgrefs on posts (msg_ref);
+                CREATE INDEX IF NOT EXISTS messages_rxseq on messages (rx_seq);
+                CREATE INDEX IF NOT EXISTS tangle_id on tangles (id);
+                CREATE INDEX IF NOT EXISTS contacts_state ON contacts (contact_id, state);
+                CREATE INDEX IF NOT EXISTS contacts_state_with_author ON contacts (author_id, contact_id, state);
                 -- add new column to posts and migrate existing data
                 ALTER TABLE posts ADD is_root boolean default false;
-                CREATE INDEX IF NOT EXISTS posts_roots on posts (is_root);
+                CREATE INDEX IF NOT EXISTS IF NOT EXISTS posts_roots on posts (is_root);
```
Comment (mplorentz) [inline Source/GoBot/ViewDatabase.swift:244]:
```suggestion
                CREATE INDEX IF NOT EXISTS posts_roots on posts (is_root);
```
looks like there's a duplicate `IF NOT EXISTS` on this line?

**Review by mplorentz — COMMENTED** [review]
@rabble bump. I don't know much about the ViewDatabase yet. I am wondering: if this SQL was failing to run because it was trying to create tables and indexes that already exist, then maybe the bug is that we are trying to apply this SQL when we shouldn't be?

Happy to merge this if you want.

**Comment by mplorentz** [comment]
Closing for staleness. Feel free to reopen.

**Pull request closed without merging**